src/utils/get-fastest-provider.ts

```diff
@@ -16,10 +15,11 @@ function getHandler(networkId: number | string) {
   return new RPCHandler(config as HandlerConstructorConfig);
 }
 
-export async function getFastestProvider(networkId: number | string): Promise<JsonRpcProvider> {
+export async function getFastestProvider(networkId: number | string): Promise<Provider> {
   try {
     const handler = getHandler(networkId);
-    return await handler.getFastestRpcProvider();
+    const provider = await handler.getFastestRpcProvider();
+    return provider as unknown as Provider;
```

Comment:
It's probably safer if you extract the `connection.url` and create a new provider. The issue is that `permit-generation` uses `ethers@v6` and `rpc-handler` uses `ethers@v5` so there are a lot of differences between the two providers.

The downside to that is we lose the benefits of the rpc-handler `Proxy`. We had started discussing it [here](https://github.com/ubiquibot/permit-generation/issues/45#issuecomment-2309567936). `viem` is off the table obviously but in terms of what to do about this repo whether to downgrade or update `rpc-handler` to support both versions, lmk

Reply:
I think we should always use ethers v5 until llms catch up with v6

Reply:
> It's probably safer if you extract the `connection.url` and create a new provider

done